docs/content/development/extensions-contrib/thrift.md

```diff
@@ -0,0 +1,104 @@
+---
+layout: doc_page
+---
+
+# Thrift
+
+To use this extension, make sure to [include](../../operations/including-extensions.html) `druid-thrift-extensions`.
+
+This extension enables Druid to ingest thrift compact data online (`ByteBuffer`) and offline (SequenceFile of type `<Writable, BytesWritable>` or LzoThriftBlock File).
+
+You may want to use another version of thrift, change the dependency in pom and compile yourself.
+
+## Thrift Parser
+
+
+| Field       | Type        | Description                              | Required |
+| ----------- | ----------- | ---------------------------------------- | -------- |
+| type        | String      | This should say `thrift`                 | yes      |
+| parseSpec   | JSON Object | Specifies the timestamp and dimensions of the data. Should be a Json parseSpec. | yes      |
+| thriftJar   | String      | path of thrift jar, if not provided, it will try to find the thrift class in classpath. Thrift jar in batch ingestion should be uploaded to HDFS first and configure `jobProperties` with `"tmpjars":"/path/to/your/thrift.jar"` | no       |
```

Comment:
Adding thrift jar to middle manager's class path will indeed work for batch ingestion, but an HDFS or some path is much more flexible. You don't have to put a jar into each nodes. Maybe pull from somewhere by http is the best. In this way, batch and realtime ways are unified. I should try to make it.